plugins/woocommerce-admin/client/products/sections/images-section.tsx

```diff
@@ -0,0 +1,191 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Link,
+	useFormContext,
+	MediaUploader,
+	ImageGallery,
+	ImageGalleryItem,
+} from '@woocommerce/components';
+import { Card, CardBody, DropZone } from '@wordpress/components';
+import { recordEvent } from '@woocommerce/tracks';
+import { useState } from '@wordpress/element';
+import { Product } from '@woocommerce/data';
+import classnames from 'classnames';
+import { Icon, trash } from '@wordpress/icons';
+import { MediaItem } from '@wordpress/media-utils';
+
+/**
+ * Internal dependencies
+ */
+import { ProductSectionLayout } from '../layout/product-section-layout';
+import DragAndDrop from '../images/drag-and-drop.svg';
+import './images-section.scss';
+
+type Image = MediaItem & {
+	src: string;
+};
+
+export const ImagesSection: React.FC = () => {
+	const { getInputProps, setValue } = useFormContext< Product >();
+	const images = ( getInputProps( 'images' ).value as Image[] ) || [];
+	const [ showRemoveZone, setShowRemoveZone ] = useState( false );
+	const [ imageIdToRemove, setImageIdToRemove ] = useState< number >( 0 );
+
+	const getUniqueImages = ( files: Image[] ) => {
+		if ( ! files ) {
+			return [];
+		}
+		files.forEach( ( image: Image ) => {
+			if (
+				image.id &&
+				images.find( ( file ) => file.id === image.id ) === undefined
+			) {
+				images.push( image );
+			}
+		} );
+		setValue( 'images', images );
+		return images;
+	};
+
+	const toggleRemoveZone = () => {
+		setShowRemoveZone( ! showRemoveZone );
+	};
+
+	const setImageToRemove = ( image: string ) => {
```

Comment:
Instead of setting the image to remove later, could we just `setValue( 'images', newImages );` here after filtering out the removed?

Reply:
Yeah, that's a good idea, and it was my first thought too. The problem I found was that it would remove the image before triggering the `onDragEnd` [here](https://github.com/woocommerce/woocommerce/pull/34769/files#diff-2d3cea1f37b6b33abf26b2628e0db24cc475409ebd4bc1903dec1d3be2e59a1bR107).
That method (among other things) removes the class `is-draggin` [in the `Sortable` component](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/components/src/sortable/sortable.tsx). So it would create a few oddities.